### DISCUSSION TITLE
How to Get Role's Permissions via relations?

### DISCUSSION THREAD

**Discussion by SaeedSoltoon** [spatie/laravel-permission#1872] [Q&A] [answered]
Hi, when I call this, I get empty result from role.

`public function getAllUsersWithAllTheirRolesAndPermissions(): Collection {`

`        return User::with(['roles:name', 'roles.permissions'])->get();`

`}`

   
And this is my table data.
![Screenshot from 2021-09-24 16-23-16](https://user-images.githubusercontent.com/4473680/134677440-27bd10ff-c24d-4324-8b47-ccff3a8d09d5.png)

and this is the result of running controller.

`{
    "status": 200,
    "data": [
        {
            "id": 1,
            "first_name": "super",
            "last_name": "admin",
            "phone_number": null,
            "email": "superadmin@test.com",
            "email_verified_at": "2021-09-22T11:39:13.000000Z",
            "is_active": true,
            "language": "en",
            "deleted_at": null,
            "created_at": "2021-09-22T11:39:13.000000Z",
            "updated_at": "2021-09-22T11:39:13.000000Z",
            "roles": [
                {
                    "name": "superadmin",
                    "permissions": []
                }
            ]
        },
        {
            "id": 2,
            "first_name": "test",
            "last_name": "test",
            "phone_number": "123",
            "email": "test@test.com",
            "email_verified_at": null,
            "is_active": true,
            "language": "en",
            "deleted_at": null,
            "created_at": "2021-09-22T12:02:46.000000Z",
            "updated_at": "2021-09-22T12:02:46.000000Z",
            "roles": [
                {
                    "name": "agent",
                    "permissions": []
                }
            ]
        }
    ]
}`

Thanks.

**Comment by erikn69** [comment, marked as answer]
Maybe because you are removing id in roles, and that's why laravel can't build the relationship

**Reply by SaeedSoltoon** [reply]
Thanks for your  reply. this is my role class, and i make changes in config file to point to this class.
I didnt remove anything. i just started from scratch.
```
namespace App\Models;
use Spatie\Permission\Models\Role as SpatieRole;
class Role extends SpatieRole
{
    protected $hidden = ['pivot'];
    public $guard_name = 'web';
}

```
what's wrong with it?

**Reply by erikn69** [reply]
>return User::with(['roles:name', 'roles.permissions'])->get();

Here you are removing the id, id is used for relation loads, must be `roles:id,name`

**Reply by SaeedSoltoon** [reply]
Thanks.
